sw/device/lib/dif/dif_otp_ctrl.h

```diff
@@ -0,0 +1,773 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_LIB_DIF_DIF_OTP_CTRL_H_
+#define OPENTITAN_SW_DEVICE_LIB_DIF_DIF_OTP_CTRL_H_
+
+/**
+ * @file
+ * @brief <a href="/hw/ip/otp_ctrl/doc/">OTP</a> Device Interface Functions
+ */
+
+#include <stdint.h>
+
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_warn_unused_result.h"
+
+// Header Extern Guard (so header can be used from C and C++)
+#ifdef __cplusplus
+extern "C" {
+#endif  // __cplusplus
+
+/**
+ * A toggle state: enabled, or disabled.
+ *
+ * This enum may be used instead of a `bool` when describing an enabled/disabled
+ * state.
+ */
+typedef enum dif_otp_ctrl_toggle {
+  /*
+   * The "enabled" state.
+   */
+  kDifOtpCtrlToggleEnabled,
+  /**
+   * The "disabled" state.
+   */
+  kDifOtpCtrlToggleDisabled,
+} dif_otp_ctrl_toggle_t;
+
+/**
+ * A partition within OTP memory.
+ */
+typedef enum dif_otp_ctrl_partition {
+  /**
+   * The creator software configuration area.
+   *
+   * This partition contains device-specific calibration data.
+   */
+  kDifOtpCtrlPartitionCreatorSwCfg,
+  /**
+   * The owner software configuration area.
+   *
+   * This partition contains data to e.g. enable ROM hardening features.
+   */
+  kDifOtpCtrlPartitionOwnerSwCfg,
+  /**
+   * The hardware configuration area.
+   */
+  kDifOtpCtrlPartitionHwCfg,
+  /**
+   * The device lifecycle area.
+   */
+  kDifOtpCtrlPartitionLifeCycle,
```

Comment:
This partition has been moved to the end of the address map. See https://github.com/lowRISC/opentitan/pull/3456.

Reply:
The order of these enum constants isn't supposed to reflect anything about the layout of the partitions in the address map (the DIF pretends each partition is isolated).